### ISSUE TITLE
Consider including "core" ltest macros

### ISSUE THREAD

**Issue by yurrriq** [lfe/lfe#296]
I'm conflicted about this, but I've introduced some redundancy when writing `clj-tests.lfe`. It might be nice to include the basic macros in LFE proper, but I also appreciate the separate release cycles. Perhaps we could use some rebar3 hacks to depend on ltest in the test profile in LFE and compile it manually, instead of with lfe-compile via hooks. I may pursue that myself but wanted to open this issue for discussion.

**Comment by rvirding** [comment]
Which ones for example? Should we include them as predefined or in an include file you can do `include-lib`? It they are "test macros" then i would prefer the second option.

**Comment by yurrriq** [comment]
Most of [`test/ltest-macros.lfe`](https://github.com/rvirding/lfe/blob/develop/test/ltest-macros.lfe) is redundant. If we decide to include the rest in LFE proper, we ought to move the file to `include/`. I'm wondering too if we might want to include ltest as a sibling app to LFE, kind of like how eunit lives alongside stdlib, kernel, et al. in OTP releases. Perhaps the rebar3 profile hack is a better approach...

**Comment by yurrriq** [comment]
It's also possible that depending on ltest in the test profile as-is will work out fine as rebar3 might find the local LFE app on the code path already and skip downloading the one bundled with lfe-compile. I think some experiments are in order, but I'm glad we have this space for discussion now.

**Comment by oubiwann** [comment]
So this was actually the first lib I considered bringing into LFE after the 2014 LFE mini-summit. 
- https://github.com/rvirding/lfe/pull/64

Related to these two:
- https://github.com/rvirding/lfe/issues/108
- https://github.com/rvirding/lfe/issues/107

As such, if we do take this approach, I would like to pick that work back up.

However, since that time, I've come around on my views somewhat ... I don't believe this is the "Erlang Way". The recent updates I've done to the LFE Docker repo has really brought this home for me, too. And I believe this is one of those points that @rvirding had tried to convey to me in the past, but didn't quite sink in: we should do LFE releases. As in the way OTP does releases.

There's a fine conceptual line to draw here: what is core to LFE? what supports LFE is such a way that it makes sense to include as part of LFE itself? What is or should be a library in its own right? By analogy, Erlang/OTP has answered that for us in eunit, should we chose to take that path.

The "Erlang Way" is not to bring desired functionality into the core `erlang` system, but rather to include it in an Erlang/OTP release. This is exactly what eunit is -- it is its own library with its own version, etc. 

Following this example, I really think we should consider starting to make LFE releases -- start small, using one or two libraries, package them with `lfe` and offer them as a download, etc. See how that goes. Iron out process and release issues. As more libraries are deemed ready for inclusion or generally useful by the community, they can be added to the release. We can also document our release process such that anyone can create their own releases, with as much or as little of the community libraries as they prefer.

This would not only provide the same familiar approach as used by the Erlang world, it would also make creating specialized (and general), pre-packaged development environments and VMs wonderfully trivial.

**Comment by yurrriq** [comment]
:+1::+1: to the idea of LFE releases. It seems like my thinking was headed in that general direction, but you really nailed it, @oubiwann.

**Comment by yurrriq** [comment]
As far as starting small goes, how about just ltest or ltest and lutil?

**Comment by oubiwann** [comment]
Cool -- I've created ticket #300 to track that work -- but I've just created it! I don't have the time to tackle that undertaking right now. However, I'm definitely available to offer comments, test out proposed processes, etc.

**Comment by rvirding** [comment]
The issue #300 is the one we need to work before we can solve this one.

**Comment by yurrriq** [comment]
Closing this in favor of #300. I think `ltest` should be included as one of the bundled applications (if not the only one) in this first release.
